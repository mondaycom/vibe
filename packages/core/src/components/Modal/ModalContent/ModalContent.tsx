import React, { forwardRef } from "react";
import { getTestId } from "../../../tests/test-ids-utils";
import { ComponentDefaultTestId } from "../../../tests/constants";
import { type ModalContentProps } from "./ModalContent.types";
import { useModal } from "../context/ModalContext";

const ModalContent = forwardRef(
  (
    { children, className, id, "data-testid": dataTestId }: ModalContentProps,
    ref: React.ForwardedRef<HTMLDivElement>
  ) => {
    const { autoFocus } = useModal();
    return (
      <div
        ref={ref}
        className={className}
        id={id}
        data-testid={dataTestId || getTestId(ComponentDefaultTestId.MODAL_NEXT_CONTENT, id)}
        data-autofocus-inside={autoFocus}
      >
        {children}
      </div>
    );
  }
);

export default ModalContent;
