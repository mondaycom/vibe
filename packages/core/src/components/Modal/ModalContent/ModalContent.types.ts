import React from "react";
import { VibeComponentProps } from "../../../types";

export interface ModalContentProps extends VibeComponentProps {
  /**
   * The main content of the modal.
   */
  children?: React.ReactNode;
}
