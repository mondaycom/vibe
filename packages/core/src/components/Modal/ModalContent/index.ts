export { default as ModalContent } from "./ModalContent";
export { type ModalContentProps } from "./ModalContent.types";
