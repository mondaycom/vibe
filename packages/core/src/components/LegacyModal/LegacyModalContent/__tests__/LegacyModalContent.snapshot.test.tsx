import { describe, it, expect } from "vitest";
import React from "react";
import renderer from "react-test-renderer";
import ModalContent from "../LegacyModalContent";

describe("ModalContent renders correctly", () => {
  it("with empty props", () => {
    const tree = renderer.create(<ModalContent>content</ModalContent>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with children", () => {
    const tree = renderer
      .create(
        <ModalContent>
          <div>children</div>
        </ModalContent>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
