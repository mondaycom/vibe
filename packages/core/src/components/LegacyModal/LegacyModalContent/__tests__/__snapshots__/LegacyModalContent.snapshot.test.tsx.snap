// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalContent renders correctly with children 1`] = `
<div
  className="typography primary start text text1Normal container"
  data-testid="modal-content"
>
  <div>
    children
  </div>
</div>
`;

exports[`ModalContent renders correctly with empty props 1`] = `
<div
  className="typography primary start text text1Normal container"
  data-testid="modal-content"
>
  content
</div>
`;
